### ISSUE TITLE
longformer-large's hidden_states and last_hidden_states have different size in sequence_length

### ISSUE THREAD

**Issue by ddf62** [huggingface/transformers#15962]
## Environment info

- `transformers` version:
- Platform: 
- Python version: 3.7
- PyTorch version (GPU?): GPU
- Tensorflow version (GPU?):
- Using GPU in script?:
- Using distributed or parallel set-up in script?:

## Information

Model I am using (Bert, XLNet ...): Longformer-large

The problem arises when using: student's homework

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [1 ] my own task or dataset: (give details below): NER

## To reproduce

1. input the sequence into Longformer-Large
2. oupt the last_states.shape and last_hidden_states[-1].shape
3. find out that them have different length in sequence_length

**Comment by ddf62** [comment]
sorry,should be the hidden_states[-1].shape and last_hidden_states.shape

**Comment by patil-suraj** [comment]
Hi @ddf62 ! Could you post a shot code snippet so we can reproduce this ? Thanks!

**Comment by ddf62** [comment]
the output is:
 torch.Size([16, 1536, 1024])  torch.Size([16, 1304, 1024])

**Comment by patil-suraj** [comment]
cc @patrickvonplaten

**Comment by patrickvonplaten** [comment]
@ddf62 - Could you please post a formatted code snippet in:

```python
class MyModel(tez.Model):
```

and one that we can reproduce? What is `tez` ? This issue also doesn't seem to be a core issue of Transformers so maybe the forum https://discuss.huggingface.co/ is the better place?

**Comment by ddf62** [comment]
sorry，you can use this code and their outputs are same：

```
from transformers import AutoTokenizer, AutoModel, LongformerTokenizerFast
import torch
class MyModel(nn.Module):
    def __init__(self, freeze_bert=False, model_name='longformer-large', hidden_size=1024, num_classes=15):
        super(MyModel, self).__init__()
        self.automodel = AutoModel.from_pretrained(model_name, output_hidden_states=True, return_dict=True)

        self.fc = nn.Sequential(
            nn.Dropout(p=0.5),
            nn.Linear(hidden_size * 4, num_classes, bias=False),
        )

        if freeze_bert:
            for p in self.automodel.parameters():
                p.requires_grad = False

    def forward(self, input_ids, attn_masks):
        outputs = self.automodel(input_ids, attention_mask=attn_masks)

        print(outputs.hidden_states[-1].shape, output.last_hidden_state.shape)
        
        hidden_states = torch.cat(tuple([outputs.hidden_states[i] for i in [-1, -2, -3, -4]]),
                                  dim=-1)  # [bs, seq_len, hidden_dim*4]
        first_hidden_states = hidden_states[:, :, :]  # [bs, hidden_dim*4]
        logits = self.fc(first_hidden_states)
        return logits
```

**Comment by patrickvonplaten** [comment]
Gently pinging @ydshieh here since he is becoming our longformer expert. If you have some time it would be amazing if you could take a look here :-)

**Comment by ydshieh** [comment]
OK, added to my TODO list :)

**Comment by ydshieh** [comment]
Hi @ddf62 

Could you also include the code that you used to create an instance of `MyModel`, prepare the inputs, and pass the inputs to the model in your above code snippet , please? This way, I can run the code to get the output directly and to investigate it.

Thank you.

**Comment by ddf62** [comment]
```
from transformers import AutoTokenizer, AutoModel, LongformerTokenizerFast
import torch
from torch import nn


class MyModel(nn.Module):
    def __init__(self, freeze_bert=False, model_name='longformer-large', hidden_size=1024, num_classes=15):
        super(MyModel, self).__init__()
        self.automodel = AutoModel.from_pretrained(model_name, output_hidden_states=True, return_dict=True)

        self.fc = nn.Sequential(
            nn.Dropout(p=0.5),
            nn.Linear(hidden_size * 4, num_classes, bias=False),
        )

        if freeze_bert:
            for p in self.automodel.parameters():
                p.requires_grad = False

    def forward(self, input_ids, attn_masks):
        outputs = self.automodel(input_ids, attention_mask=attn_masks)

        print(outputs.hidden_states[-1].shape, outputs.last_hidden_state.shape)

        hidden_states = torch.cat(tuple([outputs.hidden_states[i] for i in [-1, -2, -3, -4]]),
                                  dim=-1)  # [bs, seq_len, hidden_dim*4]
        first_hidden_states = hidden_states[:, :, :]  # [bs, hidden_dim*4]
        logits = self.fc(first_hidden_states)
        return logits


model = MyModel(model_name='../model/longformer-large', num_classes=15, freeze_bert=False)
tokenizer = LongformerTokenizerFast.from_pretrained('../model/longformer-large', add_prefix_space=True)
encoding = tokenizer('80% of Americans believe seeking multiple opinions can help them make better choices, and for good reason.'.split(),
                     is_split_into_words=True,
                     # return_offsets_mapping=True,
                     truncation=True)
model(torch.tensor([encoding['input_ids']]), torch.tensor([encoding['attention_mask']]))
```
the output is :torch.Size([1, 512, 1024]) torch.Size([1, 22, 1024])
But i think the shape of hidden_states[-1] should also be [1, 22, 1024].

**Comment by ydshieh** [comment]
Hi @ddf62

- Could you run the command `transformers-cli env` and copy-and-paste its output below. We need this information in order to reproduce the issue.
- I could not find any pretrained model with the name `longformer-large`. In your code snippet, you have `'../model/longformer-large'`, which is a local file we don't have. Could you check where does your model comes from?

With `allenai/longformer-large-4096` and a dev version `version: 4.18.0.dev0`, I can't reproduce the issue. The output I got is:

```python
torch.Size([1, 22, 1024]) torch.Size([1, 22, 1024])
```

**Comment by ddf62** [comment]
```
Copy-and-paste the text below in your GitHub issue and FILL OUT the two last points.

- `transformers` version: 4.15.0
- Platform: Linux-5.4.0-92-generic-x86_64-with-debian-buster-sid
- Python version: 3.7.11
- PyTorch version (GPU?): 1.10.1 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: 3090<fill in>
- Using distributed or parallel set-up in script?: <fill in>

```
'../model/longformer-large' is the model i download from [allenai](https://huggingface.co/allenai)
/
[longformer-large-4096](https://huggingface.co/allenai/longformer-large-4096)

**Comment by ydshieh** [comment]
> ```
> Copy-and-paste the text below in your GitHub issue and FILL OUT the two last points.
> 
> - `transformers` version: 4.15.0
> - Platform: Linux-5.4.0-92-generic-x86_64-with-debian-buster-sid
> - Python version: 3.7.11
> - PyTorch version (GPU?): 1.10.1 (True)
> - Tensorflow version (GPU?): not installed (NA)
> - Flax version (CPU?/GPU?/TPU?): not installed (NA)
> - Jax version: not installed
> - JaxLib version: not installed
> - Using GPU in script?: 3090<fill in>
> - Using distributed or parallel set-up in script?: <fill in>
> ```
> 
> '../model/longformer-large' is the model i download from [allenai](https://huggingface.co/allenai) / [longformer-large-4096](https://huggingface.co/allenai/longformer-large-4096)

Thank you! I am able to reproduce the issue. This issue is already fixed in a pull request #15537 and is included in the latest version [v4.17.0](https://github.com/huggingface/transformers/releases/tag/v4.17.0).

I am going to close this issue, but don't hesitate if you have further question. Thanks!
